### ISSUE TITLE
Question about noise in routing function during inference

### ISSUE THREAD

**Issue by zzzres** [eduardzamfir/MoCE-IR#19]
Hi,

Thanks for sharing this great work and the clear implementation.

I noticed that noise is added to the routing logits in the routing function:

logits = self.gate(x) + self.freq_gate(freq_emb)
if self.training:
    loss_imp = self.importance_loss(logits.softmax(dim=-1))

noise = torch.randn_like(logits) * self.noise_std
noisy_logits = logits + noise
gating_scores = noisy_logits.softmax(dim=-1)
top_k_values, top_k_indices = torch.topk(gating_scores, self.k, dim=-1)

Since the noise is always added to the logits, it seems that the routing decisions could be stochastic during inference as well. This might lead to different expert selections for the same input across different runs.

Should the routing noise only be applied during training and be disabled during inference?

Thanks!
